packages/providers/src/core/PoapCollectionsApi/PoapCollectionsApi.ts

```diff
@@ -0,0 +1,112 @@
+   * Creates an instance of PoapCollectionsApi.
+   *
+   * @constructor
+   * @param {string} [baseUrl='https://collections.poap.tech'] - The base URL for the Collections API.
+   * If not provided, a default URL is used.
+   */
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl || DEFAULT_COLLECTIONS_BASE_URL;
+  }
+
+  /**
+   * Posts a new collection to the POAP Collections API.
+   *
+   * @public
+   * @async
+   * @param {PostCollectionsInput} postCollectionsInput - The collection data to be posted.
+   * @param {string} accessToken - The access token for authorization.
+   * @returns {Promise<CollectionResponse>} A promise that resolves with the collection response from the API.
+   */
+  public async postCollection(
+    postCollectionsInput: PostCollectionsInput,
+    accessToken: string,
+  ): Promise<CollectionResponse> {
+    return await this.secureFetch<CollectionResponse>(
+      `${this.baseUrl}/collections`,
+      {
+        method: 'POST',
+        body: JSON.stringify(postCollectionsInput),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+      accessToken,
+    );
+  }
+
+  /**
+   * Applies a update to an existing collection in the POAP Collections API.
+   *
+   * @public
+   * @async
+   * @param {PutCollectionsInput} putCollectionsInput - The put data and collection ID.
+   * @param {string} accessToken - The access token for authorization.
+   * @returns {Promise<CollectionResponse>} A promise that resolves with the updated collection response from the API.
+   */
+  public async putCollection(
+    putCollectionsInput: PutCollectionsInput,
+    accessToken: string,
+  ): Promise<CollectionResponse> {
+    return await this.secureFetch<CollectionResponse>(
+      `${this.baseUrl}/collections/${putCollectionsInput.collectionId}`,
+      {
+        method: 'PATCH',
+        body: JSON.stringify(putCollectionsInput),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+      accessToken,
+    );
+  }
+
+  /**
+   * Sends a secure HTTP request to the POAP Collections API.
+   *
+   * This method is used internally by `postCollection` and `patchCollection` to send HTTP requests.
+   *
+   * @private
+   * @async
+   * @param {string} url - The URL for the HTTP request.
+   * @param {RequestInit} options - The options for the HTTP request, including method and body.
+   * @param {string} accessToken - The access token for authorization.
+   * @returns {Promise<R>} A promise that resolves with the response from the API.
+   */
+  private async secureFetch<R>(
+    url: string,
+    options: RequestInit,
+    accessToken: string,
+  ): Promise<R> {
+    const headersWithApiKey = {
+      ...options.headers,
+      Authorization: `Bearer ${accessToken}`,
+    };
+
+    const response = await fetch(url, {
+      method: options.method,
+      body: options.body,
+      headers: headersWithApiKey,
+    });
+
+    return await response.json();
```

Comment:
When response is not a JSON this throws an exception.

Before is missing the check for the response status code to be 200 to parse the response.

Each endpoint should also check for specific response status code, as it might return a 400 with JSON body or a 410 or something else specific to each endpoint.